### ISSUE TITLE
Meeting 11th May 2020

### ISSUE THREAD

**Issue by fubuloubu** [vyperlang/vyper#1941]
Start time: 18:00 UTC (https://time.is/UTC)

Vidchat link: https://meet.google.com/nvw-nvzt-zrs

Agenda:
- [x] #1944
- [x] #1950
- [x] #1953
- [x] #1954
- [x] #1955
- [x] #1963
- [x] #1440 (Approved, but changes were made to the syntax afterwards)

**Comment by sbellem** [comment]
Hi,

 is the meeting open to "outsiders"? I am asking because I am curious about the roadmap of the project and perhaps I could help a bit.

Thanks,
Sylvain

**Comment by fubuloubu** [comment]
@sbellem yes, it is open! We try to have the meeting on time, but sometimes we end up missing it, so just keep an eye on gitter and this issue around the time it is scheduled to start for updates on date/time and vidchat links. You are free to listen, join in the discussion, or propose topics to discuss in the agenda here!

**Comment by sbellem** [comment]
Thanks @fubuloubu!

**Comment by fubuloubu** [comment]
Call in 45 mins!

**Comment by rayrrr** [comment]
Sorry I missed the meeting, busy day at work. Was it recorded by chance? Wondering on any outcomes related to https://github.com/vyperlang/vyper/issues/1440

**Comment by fubuloubu** [comment]
@rayrrr we usally don't record the meetings, they're fairly low-key, but we feed back all decisions to GitHub.

On #1440, @iamdefinitelyahuman and I bikeshedded for 20 mins and we feel like we finally arrived at a good specification. It's only a mild bastardization of the Python slice syntax, but it effectively makes a delineation between static and dynamic arrays. Probably the first step towards implementing it will be to extend `bytes[N]` and `string[N]` by adding the `calldata` min-max length check, and require the new syntax of `Bytes[M:N]` and `String[M:N]` (which is a breaking change). This will hopefully implement most of the functionality in the metaclasses that will support the more general `Array[type, M:N]` syntax, where `type` can be any type with a statically known size (at compile time). We might do this in stages, starting with basic types first, but this is currently the number 1 most-requested feature on our list for the 0.2.x-series (followed closely by `abi_decode`)
